resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -231,7 +248,115 @@ class MousePointer {
     }
 }
 
+[DSCResource()]
+class AnimationEffects {
+	#Need to verify the Animation Effects setting toggle on the Accessibility page changes when these are updated, or find the setting to update that also.
+    [DscProperty(Key)] [BinarySettingState] $SmoothScrollListBoxes = [BinarySettingState]::KeepCurrentValue
+    [DscProperty()] [BinarySettingState] $SlideOpenComboBoxes = [BinarySettingState]::KeepCurrentValue
+    [DscProperty()] [BinarySettingState] $FadeOrSlideMenusIntoView = [BinarySettingState]::KeepCurrentValue
+    [DscProperty()] [BinarySettingState] $ShowShadowsUnderMousePointer = [BinarySettingState]::KeepCurrentValue
+    [DscProperty()] [BinarySettingState] $FadeOrSlideToolTipsIntoView = [BinarySettingState]::KeepCurrentValue
+    [DscProperty()] [BinarySettingState] $FadeOutMenuItemsAfterClicking = [BinarySettingState]::KeepCurrentValue
+    [DscProperty()] [BinarySettingState] $ShowShadowsUnderWindows = [BinarySettingState]::KeepCurrentValue
+    [DscProperty(NotConfigurable)] [AnimationEffects] $currentState
+
+    [AnimationEffects] Get() {
+        $this.currentState = [AnimationEffects]::new()
+
+		$this.currentState = [BinarySettingState]::Disabled
+
+		foreach ($enum in [AnimationEffectsSettings].GetEnumNames()) {
+			$thisState = Get-AnimationState $enum
+			if ($thisState -eq [BinarySettingState]::Enabled) {
+				$this.currentState = [BinarySettingState]::Enabled
+			}
+		}
+
+		return $this.currentState
+	}
+
+    [bool] Test() {
+		$this.currentState = $this.Get()
+		if ($this.SmoothScrollListBoxes -ne [BinarySettingState]::KeepCurrentValue -or
+		$this.SmoothScrollListBoxes -ne $this.currentState.SmoothScrollListBoxes) {
+
+			return $false
+		}
+		
+		else if ($this.SlideOpenComboBoxes -ne [BinarySettingState]::KeepCurrentValue -or
+		$this.SlideOpenComboBoxes -ne $this.currentState.SlideOpenComboBoxes) {
+
+			return $false
+		}

```

Comment:
I would just do a bunch of if statements and at the end return true if none of the conditions were satisfied.